controllers/cluster_controller.go

```diff
@@ -357,10 +359,17 @@ func (c *clusterDescendants) descendantNames() string {
 	if len(workerMachineNames) > 0 {
 		descendants = append(descendants, "Worker machines: "+strings.Join(workerMachineNames, ","))
 	}
+	machinePoolNames := make([]string, len(c.machinePools.Items))
+	for i, machinePool := range c.machinePools.Items {
+		machinePoolNames[i] = machinePool.Name
+	}
+	if len(machinePoolNames) > 0 {
+		descendants = append(descendants, "Machine pools: "+strings.Join(machinePoolNames, ","))
+	}
```

Comment:
This should likely be wrapped with a check to see if the feature gate for MachinePools is enabled, similar to what we are doing here: https://github.com/kubernetes-sigs/cluster-api/blob/2bb56ed2eb2a91cb59b740cb286992952da3dfed/bootstrap/util/configowner.go#L95-L100